### PR TITLE
Update documentation

### LINKED ISSUE #160: [PE-D][Tester E] Minor formatting error in UG
It is possible that "Note on returning to the origin view" bullet point was included by mistake in the list of all commands as a sub-command? It appears to be inconsistent

Regardless, even if it is intended, maybe the bullet-point style should be different from the previous level?

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: tahsinhasem/ped#2

### LINKED ISSUE #163: [PE-D][Tester E] Minor Documentation clarification needed for navigation of commands
Perhaps what you mean is "you are able to navigate through your past successful command and **current command** by using the `UP` and `DOWN` arrow keys. 

As if I type two successful commands and one wrong, I am still able to navigate between all 3. Example: `help`, `list`, `welp`. I can press up and down commands to navigate through all!

-------------
Labels: `type.DocumentationBug` `severity.VeryLow`
original: tahsinhasem/ped#3

### LINKED ISSUE #167: [PE-D][Tester E] Not showing changes after nuscontacts.json is updated.
This is probably an edge case test. However, after I updated the `data/nuscontacts.json` file and added a tag to John Doe, and then proceeded with `list` command, it did not show the new tag for John Doe, which I think is an expected feature? The tag would also not appear when I add a new person to nuscontact manager. The ` data/nuscontacts.json file` also did not seem to register the new person after I added him/her.

When I closed the app, all changes I made to the` data/nuscontacts.json file` was erased.

-------------
Labels: `type.FeatureFlaw` `severity.Low`
original: tahsinhasem/ped#4

### LINKED ISSUE #171: [PE-D][Tester E] No definition of unambiguous prefix provided
No definition of `unambiguous prefix` provided

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: tahsinhasem/ped#5

### LINKED ISSUE #178: [PE-D][Tester D] Only 1 error message is shown even if there are multiple
This command should show the error messages:
The following is one of the invalid fields(s). Fix the issue, or append f/ to ignore the warning.
Phone numbers should only contain numbers, and it should be at least 3 digits long.

and

Multiple values specified for the following single-valued field(s): a/

but only the 2nd one is shown

-------------
Labels: `type.FunctionalityBug` `severity.Medium`
original: sindhurajain/ped#3

### LINKED ISSUE #183: [PE-D][Tester D] Confusing use of ellipses in user guide
In the add command section of the user guide:

It's better to write out the full add command like add f/ n/Alice e/alice@gmail.com r/student c/cs2103T instead of add f/ n/Alice ... as the ellipses is supposed to represent fields that can be used multiple times including zero times, which is not how it is used here

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: sindhurajain/ped#7

### LINKED ISSUE #190: [PE-D][Tester D] f/ behaviour differs from the user guide
Th user guide, under the add command feature, says that f/ will bypass some validation rules, but it is not clear which rules f/ will bypass out of all the validation rules:

However, f/ does not bypass the empty name error:

so its behaviour differs from what is specified in the user guide. To solve this, I think you can either 
1. specify in the user guide which validation rules f/ will bypass
or 
2. allow f/ to bypass the empty name error

-------------
Labels: `type.FunctionalityBug` `severity.Medium`
original: sindhurajain/ped#14

### LINKED ISSUE #194: [PE-D][Tester B] legitimate escape sequences not well defined in ug
The UG states that invalid escape sequences should not be entered, and the program would guess. I'm assuming that the program treats \s above as \\s, thus leading to the same result, and escapes the character \. However isn't \s a correct sequence for <space>. should be more clear what is allowed and not allowed in the guide.

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: ChrysanthemumT/ped#5

### PR DESCRIPTION
**Pull request by E0735389**
Closes #171
Closes #183
Closes #163
Closes #194
Closes #190
Closes #178
Closes #167
Closes #160

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -59,6 +59,8 @@ NUSContacts data are saved in the hard disk automatically after any command that
 
 NUSContacts data are saved automatically as a JSON file `[JAR file location]/data/nuscontacts.json`. Advanced users are welcome to update data directly by editing that data file.
 
+You should only update the data when the program is closed, otherwise there is a risk of data loss as the program may not detect the changes.
```
Comment (ChillinRage) [inline docs/UserGuide.md:62]: I think better to specify “modify the data file when…”

File: docs/UserGuide.md [hunk 2]
```diff
@@ -363,6 +368,55 @@ For example:
...
+
+A string `s` is a **prefix** of another string `t` if `s` can be obtained by removing zero or more characters at the end
+of `t`. For example, "ab" is a prefix of "abcd", but it's not a prefix of "acbd".
+
+A prefix for a command is **unambiguous** if there is exactly one command that it is a prefix of,
+and a prefix is **ambiguous** if there are at least two commands it is a prefix of.
+For example, since there are two commands `edit` and `exit`, the string "e" would be an **ambiguous** prefix, since it
+is a prefix of two distinct commands.
+
+Completely similarly, a prefix for a role is unambiguous if there is exactly one role that it is a prefix of.
```
Comment (ChillinRage) [inline docs/UserGuide.md:415]: Weird phrasing “completely similarly”. Can remove “completely”.
C

**Review by ChillinRage — COMMENTED** [review]
NP here except some nitpicks.

**Review by ChillinRage — APPROVED** [review]
